digitalearthau/coherence.py

```diff
@@ -40,63 +47,121 @@ def main(expressions, archive_locationless, debug, check_locationless, check_anc
     you've synced the ancestor collections too.
 
     (the sync tool cannot do these checks "live", as we don't know how many locations there are of a dataset
-    until the entire folder has been synced, and ideally the ancestors synced too.
+    until the entire folder has been synced, and ideally the ancestors synced too.)
     TODO: This could be merged into it as a post-processing step, although it's less safe than sync if
-    the index is being updated concurrently by another)
+    TODO: the index is being updated concurrently by another
     """
+    global _siblings_count
     uiutil.init_logging()
-    with Datacube() as dc:
+    config_file = test_dc_config if test_dc_config else ''
+    with Datacube(config=config_file) as dc:
         _LOG.info('query', query=expressions)
         count = 0
         archive_count = 0
+        locationless_count = 0
         for dataset in dc.index.datasets.search(**expressions):
             count += 1
             # Archive if it has no locations.
             # (the sync tool removes locations that don't exist anymore on disk,
             # but can't archive datasets as another path may be added later during the sync)
             if check_locationless or archive_locationless:
                 if dataset.uris is not None and len(dataset.uris) == 0:
-                    _LOG.info("dataset.locationless", dataset_id=str(dataset.id))
+                    locationless_count += 1
+
                     if archive_locationless:
                         dc.index.datasets.archive([dataset.id])
                         archive_count += 1
+                        _LOG.info("locationless_dataset_id.archived", dataset_id=str(dataset.id))
+                    else:
+                        _LOG.info("locationless_dataset_id", dataset_id=str(dataset.id))
 
             # If an ancestor is archived, it may have been replaced. This one may need
             # to be reprocessed too.
-            if check_ancestors:
-                _check_ancestors(check_siblings, dc, dataset)
+            if check_ancestors or archive_siblings or check_siblings:
+                archive_count += _check_ancestors(check_ancestors, check_siblings, archive_siblings, dc, dataset)
+
+        _LOG.info("coherence.finish",
+                  datasets_count=count,
+                  locationless_count=locationless_count,
+                  siblings_count=_siblings_count,
+                  archived_count=archive_count)
+
 
-        _LOG.info("coherence.finish", count=count, archive_count=archive_count)
+def _archive_dataset_ids(dc, ds_id, sibling_ids):
+    # sibling/s indexed_time dictionary
+    sb_itime_dict = dict()
 
+    # Count of dataset id's archived
+    archive_count = 0
 
-def _check_ancestors(check_siblings: bool, dc: Datacube, dataset: Dataset):
+    # dataset indexed time
+    ds_itime = dc.index.datasets.get(ds_id).indexed_time
+    for key, s_id in enumerate(sibling_ids):
```

Comment:
It's possible to iterate directly over `sibling_ids`, the `key` and `enumerate()` aren't needed here.

Reply:
This condition is replaced by new function.